src/v/storage/index_state.cc

```diff
@@ -64,7 +64,7 @@ bool index_state::maybe_index(
         // We know that a segment cannot be > 4GB
         add_entry(
           batch_base_offset() - base_offset(),
-          last_timestamp() - base_timestamp(),
+          std::max(last_timestamp() - base_timestamp(), int64_t{0}),
```

Comment:
subtle. 

I was thinking base_timestamp() as the bottom time, but you are right, it has to be 0.